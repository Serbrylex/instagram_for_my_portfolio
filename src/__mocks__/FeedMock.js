const FeedMock = [
  {
    user_id: 11,
    username: 'beforea',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/orig_203260_vpYx4nF.jpg',
    post_id: 17,
    description: 'want cry you the a I skin have But But wish weirdo happy belong creep',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/atrowhat_i4wcO1X.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/orig_203260_2yXnAAV.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/GettyImages_qmfgiH5.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10
    ],
    all_likes: 10,
    all_comments: 10
  },
  {
    user_id: 12,
    username: 'creepYoure',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/MyPortafolio_Q3mcJG8.png',
    post_id: 18,
    description: 'a belong a angel a you am What was like \ncreep \nbeautiful a world was',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/MyPortafolio_P4GtQ4f.png',
      'http://127.0.0.1:8000/media/posts/pictures/maxresdefault_JYTjTzL.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11
    ],
    all_likes: 11,
    all_comments: 11
  },
  {
    user_id: 13,
    username: 'aroundangel',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/',
    post_id: 19,
    description: 'perfect the a fuckin perfect I the run was was so I makes a \nhere',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/instagram_8K1lCZx.png',
      'http://127.0.0.1:8000/media/posts/pictures/Proyect_k1K5PC6.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 13,
    username: 'aroundangel',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/',
    post_id: 20,
    description: 'When so weirdo weirdo run \ncreep belong I am was I skin hell I hell',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/Logo_q10b3fS.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 13,
    username: 'aroundangel',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/',
    post_id: 21,
    description: 'hell Youre want I weirdo dont Whatever wish so \ncare creep What \nhere happy around',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/Anon_cDLpk7h.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/maxresdefault_H7IDg0J.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 13,
    username: 'aroundangel',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/',
    post_id: 22,
    description: 'was I perfect Youre feather wanna here you here special here the She hell wish',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/instagram_ZVIJlja.png',
      'http://127.0.0.1:8000/media/posts/pictures/Proyect_4KAwHbP.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 13,
    username: 'aroundangel',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/',
    post_id: 23,
    description: 'world a want you out Im fuckin Youre a so Im \nspecial I want Youre',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/Sin-t√≠tulo-4_tDoYunM.png',
      'http://127.0.0.1:8000/media/posts/pictures/orig_203260_gW5w1Wh.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/instagram_E8ZeFdQ.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 14,
    username: 'sowant',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/Logo.png',
    post_id: 24,
    description: 'What fuckin doin I here belong run weirdo run Youre skin I Shes were here',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/MyPortafolio_HrWHfeZ.png',
      'http://127.0.0.1:8000/media/posts/pictures/MainImage_tK3bmWj.png',
      'http://127.0.0.1:8000/media/posts/pictures/Anon_y5FD8tO.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 14,
    username: 'sowant',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/Logo.png',
    post_id: 25,
    description: 'wish dont \nbeautiful But makes the belong So door hell Youre angel She I here',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/o1241679_IW56RkR.png',
      'http://127.0.0.1:8000/media/posts/pictures/Cursos_X4L64tk.png',
      'http://127.0.0.1:8000/media/posts/pictures/GettyImages_QKilziA.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 14,
    username: 'sowant',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/Logo.png',
    post_id: 26,
    description: 'before fuckin weirdo was I wish I a eye hell I was a I She',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/Proyect_Gb0y2uP.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 14,
    username: 'sowant',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/Logo.png',
    post_id: 27,
    description: 'you belong before dont am run want want happy I wish belong control me \nspecial',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/atrowhat_7Q5ouCp.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/o1241679_Fjw8O12.png',
      'http://127.0.0.1:8000/media/posts/pictures/Anon_GaEdm4I.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 14,
    username: 'sowant',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/Logo.png',
    post_id: 28,
    description: 'I I run have weirdo belong the dont She run a body not special Im',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/GettyImages_vmojJcc.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/Anon_FwPxdPA.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/GettyImages_PfGWrs3.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 15,
    username: 'anIm',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/agujero_1KnEnoO.jpg',
    post_id: 29,
    description: 'run Youre weirdo running world weirdo here belong belong float Im But like Shes I',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/MyPortafolio_8LNJGbi.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 15,
    username: 'anIm',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/agujero_1KnEnoO.jpg',
    post_id: 30,
    description: 'run I me the \nspecial a When I But you want a dont special weirdo',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/o1241679_GTmhzOD.png',
      'http://127.0.0.1:8000/media/posts/pictures/agujero_L1xkXym.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 15,
    username: 'anIm',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/agujero_1KnEnoO.jpg',
    post_id: 31,
    description: 'cry you happy a Youre\njust But so the But a dont the What you want',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/o1241679_xGHdI6T.png',
      'http://127.0.0.1:8000/media/posts/pictures/MainImage_Zfp5Bs8.png',
      'http://127.0.0.1:8000/media/posts/pictures/MyPortafolio_jNQ8C2n.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 15,
    username: 'anIm',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/agujero_1KnEnoO.jpg',
    post_id: 32,
    description: 'was belong In it \nWhatever a you I float here here run were here was',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/GettyImages_q6IpajF.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 15,
    username: 'anIm',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/agujero_1KnEnoO.jpg',
    post_id: 33,
    description: 'I here in me it Youre feather fuckin Im weirdo perfect float want I Im',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/maxresdefault_8pmn5BF.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 16,
    username: 'specialI',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/orig_203260_fgkgSon.jpg',
    post_id: 34,
    description: 'dont I I dont a around world I you want I am I want float',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/maxresdefault_Zn4dM2s.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/agujero_mIv5NA7.jpg'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 16,
    username: 'specialI',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/orig_203260_fgkgSon.jpg',
    post_id: 35,
    description: 'creep run wanna dont When I I hurts fuckin hell perfect \ncare \nWhatever am I',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/atrowhat_Ky8vXqN.jpg',
      'http://127.0.0.1:8000/media/posts/pictures/Logo_caU82Wr.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  },
  {
    user_id: 16,
    username: 'specialI',
    created: '1 month ago',
    picture: 'http://127.0.0.1:8000/media/users/pictures/orig_203260_fgkgSon.jpg',
    post_id: 36,
    description: 'Whatever So Run to was skin I creep I look I Im around I I',
    images: [
      'http://127.0.0.1:8000/media/posts/pictures/Cursos_3zf8Mcd.png',
      'http://127.0.0.1:8000/media/posts/pictures/instagram_WnxslhN.png'
    ],
    users_liked: [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
      9,
      10,
      11,
      12,
      13,
      14,
      15,
      16
    ],
    all_likes: 16,
    all_comments: 16
  }
]

export default FeedMock