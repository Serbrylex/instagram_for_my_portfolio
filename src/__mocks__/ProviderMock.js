import React from 'react'
import { compose, createStore } from 'redux'
import { BrowserRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { createBrowserHistory } from 'history'

import rootReducer from '../reducers/rootReducer'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(
	rootReducer,
	composeEnhancers()
)
const history = createBrowserHistory()

const ProviderMock = props => (
	<Provider store={store}>
		<BrowserRouter history={history}>
			{props.children}		
		</BrowserRouter>
	</Provider>
)

export default ProviderMock