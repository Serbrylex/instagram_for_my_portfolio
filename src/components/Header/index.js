// React
import { useSelector } from 'react-redux'
import { useLocation, useNavigate } from 'react-router-dom'
import React from 'react'

// Assets
import {
	HeaderContainer, Title, ListIcons, ProfileData, 
	Username, Link, ImageProfile, FormContainer, Input,
	LinkSearch
} from './style' 

import imageTest from '../../assets/images/agujero-del-tiempo.jpg'
 
// Icons
import { CgAddR } from 'react-icons/cg'
import { 
	AiOutlineSearch, AiOutlineLock, AiOutlineUnlock, AiFillHome, AiOutlineHome
} from 'react-icons/ai'
import { MdKeyboardArrowDown } from 'react-icons/md'
import { GiHamburgerMenu } from 'react-icons/gi'
import { ImSearch } from 'react-icons/im'

// Hooks
import { useInputValue } from '../../hooks/useInputValue'

 
const Header = ({ setShowUserMenu }) => { 

	// Context
	const { theme, url } = useSelector(store => store.preference)
	const user = useSelector(store => store.user)
	const searching = useInputValue('Search')

	// Url location
	let site = useLocation();
	site = site.pathname
	const navigate = useNavigate()
	
	// Variables
	const size = '22px'
	const color = theme === 'light' ? 'black' : 'white'

	let title = 'Instagram'		
	// Because if footer is showed, we don't need this
	let show_elements = window.screen.width > 400

 	return(
 		<HeaderContainer theme={theme}>
 			{!site.includes('/profile/') && 			
	 			<Title>{title}</Title> 			
	 		}
 			{site.includes('/profile/') && 				
				<ProfileData>
					{user.user.is_public ?
						<AiOutlineLock size={size} color={color}/> :
						<AiOutlineUnlock size={size} color={color}/>
					}
					<Username>{user.user?.username ? user.user?.username : 'Unknown'}</Username>
					<MdKeyboardArrowDown size={size} onClick={()=>console.log('Desplega la wea')} color={color}/>
				</ProfileData> 				
 			}
 			<FormContainer>
 				<Input {...searching} onSubmit={e => console.log('hihi')}/>
 				<LinkSearch to={`/search/${searching.value}`} id='search'>														
					<AiOutlineSearch size={size} color={color} />
				</LinkSearch>
 			</FormContainer>
	 		<ListIcons>
				{user.isAuth && show_elements && 
					<>
						<Link to='/' id='feed'>							
							<AiFillHome size={size} color={color} />							
						</Link>
						<Link to='/new-post' id='new_post'>
							<CgAddR size={size} color={color}/>
						</Link>					
					</>
				}
	 			<Link to={user.user?.username ? `/profile/${user.user.username}` : '/login/'} id='user_picture' onClick={()=>navigate(`/profile/${user.user.username}`)}>
					<ImageProfile src={user.user.profile?.picture ? `${url}${user.user.profile.picture}` : imageTest} alt="Profile Image" border={''}/>
				</Link>
	 			{site !== '/' && !site.includes('search') &&
	 				<GiHamburgerMenu size={size} onClick={()=>setShowUserMenu(true)} color={color} id='hamburguer_button' />
	 			}
	 		</ListIcons>
 		</HeaderContainer>
 	)
 }

export default Header