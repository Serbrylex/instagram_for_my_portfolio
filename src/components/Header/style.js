import styled, { css } from 'styled-components'

import { Link as Linked } from 'react-router-dom'

export const HeaderContainer = styled.div`
	width: 100%;
	height: 60px;	 
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	align-items: center;
	justify-content: space-between;
	padding: 5px 5%;
	border-bottom: 1px solid #dbdbdb;
	position: absolute;
	top: 0;
	left: 0;
	right: 0;
	z-index: 100;	

	${props => props.theme === 'light' && css`
		background-color: white;
		color: black;
	`}

	${props => props.theme === 'dark' && css`
		background-color: #1F1F1F;
		color: white;
	`}
`
 
export const Title = styled.h1`
	width: auto;
	font-size: 25px;
	font-family: 'ZCOOL KuaiLe', cursive;
	cursor: pointer;
`

export const ListIcons = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-evenly;		
	width: 80%;
	height: 100%;
	margin-left: auto;
`

export const ProfileData = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-evenly;
	width: 80%;
`

export const Username = styled.div`
	width: auto;
`

export const Link = styled(Linked)`
	text-decoration: none;
	color: black;
	height: auto;
	margin: 5px;
`

export const ImageProfile = styled.img`
	width: 30px;
	height: 30px;
	border: ${props => props.border.length > 0 ? `2px solid ${props.border}` : 'none'};
	border-radius: 100%;
	object-fit: cover;
`

export const FormContainer = styled.div`	
	width: 100%;
	height: 80%;		
	display: flex;
	position: relative;		
	align-items: center;
`

export const LinkSearch = styled(Linked)`
	text-decoration: none;
	height: 22px;	
	color: black;	
	position: absolute;
	right: 2%;
	z-index: 10;	
	top: 0;
	bottom: 0;
	margin: auto 0;
`

export const Input = styled.input`
	width: 100%;
	height: 100%;
	padding: 2px 5px;
	background-color: #141414;
	color: white;
	border: none;
	border-radius: 10px;
`
