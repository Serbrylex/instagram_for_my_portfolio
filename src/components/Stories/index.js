// React
import { useEffect, useState } from 'react'
import { useNavigate } from 'react-router-dom' 
import { useSelector } from 'react-redux'

// Assets
import { 
	ListOfStories, Stori, StoriContainer, UsernameShort, ImgageStorie,
	ArrowsContainer, ListOfStoriesFor
} from './style'
 
import { AiOutlinePlus } from 'react-icons/ai'
import { IoIosArrowBack, IoIosArrowForward } from 'react-icons/io'

// Hooks
import { useGetWords } from '../../hooks/useGetWords'
 
 
const Stories = ({ add, stories, isMyProfile = false }) => {
 
	// Context
	const user = useSelector(store => store.user)
 	const { theme, url } = useSelector(store => store.preference)

	// Variables
	const history = useNavigate()
	const color = theme === 'light' ? 'black' : 'white'
	const size = '22px'
	const [storiesReference, setStoriesReference] = useState(false)
	// Esto sirver para saber en página de las Stories te encuentras,
	// lo maximo de stories por página es de 6
	const [storiesIndex, setStoriesIndex] = useState(0)

	// Language hook
 	const words = useGetWords({ component: 'stories' }) 	

 	useEffect(()=>{ 		
 		setStoriesReference(document.getElementById('list-of-stories')) 		
 	}, [])

	const handleMoveArrow = (whatDoYouWant) => {				
		let actualNotAsyncImageIndex = storiesIndex
		if (whatDoYouWant) {
			if (storiesIndex > 0) {
				actualNotAsyncImageIndex -= 1 
				setStoriesIndex(actualNotAsyncImageIndex)
			}
		} else {
			if (storiesIndex < stories.length - 1) {
				actualNotAsyncImageIndex += 1
				setStoriesIndex(actualNotAsyncImageIndex)	
			}
		}


		if (storiesReference !== false) {
			if (window.screen.width > 500) {
				storiesReference.scroll({
					left: 500 * actualNotAsyncImageIndex
				})
			} else {
				storiesReference.scroll({
					left: window.screen.width * actualNotAsyncImageIndex
				})
			}
		}		
	}


	return (
		<ListOfStoriesFor>	
			{storiesIndex !== 0 &&
				<ArrowsContainer theme={theme} orientation={'left'}>
					<IoIosArrowBack 
						size={size} 
						color={theme === 'light' ? 'white' : 'black'}
						onClick={()=>handleMoveArrow(true)}
					/>
				</ArrowsContainer>
			}
			{stories.length > 4 &&
				<ArrowsContainer theme={theme} orientation={'right'}>
					<IoIosArrowForward 
						size={size} 
						color={theme === 'light' ? 'white' : 'black'}
						onClick={()=>handleMoveArrow(false)}
					/>
				</ArrowsContainer>
			}
			<ListOfStories theme={theme} id='list-of-stories'>	
				{add === 'last' && isMyProfile &&
					<Stori onClick={()=> history('/add-storie') }>
						<StoriContainer theme={theme} style={{border: `1px solid ${color}`}}>						
							<AiOutlinePlus color={color}/>
						</StoriContainer>
						<UsernameShort theme={theme}>{words?.new_storie}</UsernameShort>
					</Stori>	
				}		
				{stories?.map((eachUserStorie, index)=>(
					<Stori key={index} onClick={()=>{history(eachUserStorie.eventAddOrGo)}}>
						<StoriContainer theme={theme}>						
							<ImgageStorie src={eachUserStorie.user.picture} alt={eachUserStorie.user.username} />
						</StoriContainer>
						<UsernameShort theme={theme}>{eachUserStorie.user.username}</UsernameShort>
					</Stori>
				))}						
			</ListOfStories>
		</ListOfStoriesFor>
	)
}

export default Stories