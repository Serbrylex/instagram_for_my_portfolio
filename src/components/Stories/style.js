import styled from 'styled-components'

// Stories --------------------------------------------
export const ListOfStories = styled.div`
	display: flex;
	width: 100vw;
	height: 100px;
	align-items: center;
	border-bottom: 1px solid grey;	
	overflow-x: scroll;
	scrollbar-width: none;
	justify-content: start;
`

export const Stori = styled.div`	
	width: 15vw;
	height: 100%;
	margin: 0 5px;
	display: flex;
	flex-direction: column;
	align-items: center;
	text-align: center;
	justify-content: center;
	text-decoration: none;	
	cursor: pointer;
`

export const StoriContainer = styled.div`
	border-radius: 100%;
	width: 50px;
	height: 50px;	
	display: flex;
	align-items: center;
	justify-content: center;	
`

export const ImgageStorie = styled.img`
	border-radius: 100%;
	width: 100%;
	height: 100%;
	object-fit: cover;
`

export const UsernameShort = styled.p`
	width: 15vw;
	color: black;
	font-size: 12px;
	font-weight: bold;
`