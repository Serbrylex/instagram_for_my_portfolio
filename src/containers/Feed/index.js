// React
import { useState, useContext, useEffect } from 'react'
import { Helmet } from 'react-helmet'
 
// Assets 
import {   
	FeedContainer, FeedImages, LoadingBottom, ImageLoading,
	ContainerButtonFor, ButtonToSearch, LinkFor
} from './style'

// Components
import Header from '../../components/Header' 
import Footer from '../../components/Footer'
import Stories from '../../components/Stories'
import Loading from '../../components/Loading'
import PostsContainer from '../../components/PostsContainer'

// Context
import UserContext from '../../context/users'
import ThemeContext from '../../context/theme'

// Hooks
import { useGetStories } from '../../hooks/useGetStories'
import { useGetPosts } from '../../hooks/useGetPosts' 

let element = undefined
 
const Feed = ({ url }) => {		
 	
 	// Context
	const { isAuth } = useContext(UserContext) 	 	
 	const { theme } = useContext(ThemeContext) 	 		
 	
	// Stories and posts
	let stories = useGetStories({ 
		token: isAuth.access_token, 
		user: isAuth.user, 
		url: url
	})	
		
	// Es el hook que va a hacer las peticiones
	const posts = useGetPosts({ 
		token: isAuth.access_token,
		user: isAuth.user, 
		url: url
	})
	

	return(
		<FeedContainer theme={theme}>
			<Helmet>
                <title>Instagram</title>
				<meta name='description' content={`Feed de instagram`} />
			</Helmet>
			<Header url={url}/>

			{posts.loading ? 
				<Loading /> :
				<>
					<Stories add='first' stories={stories} url={url} />					

					<FeedImages theme={theme}>
						{posts.posts.length > 0 ?
							<PostsContainer posts={posts.posts} url={url} /> :
							<ContainerButtonFor>
								<ButtonToSearch>No sigues a nadie, <LinkFor to='/search'>presiona aqu√≠ para buscar gente</LinkFor></ButtonToSearch>
							</ContainerButtonFor>
						}
					</FeedImages>					
				</>
			}

			<Footer />
		</FeedContainer>
	)
}
 
export default Feed