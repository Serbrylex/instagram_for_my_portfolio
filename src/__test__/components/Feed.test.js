import React from 'react'
import { mount, shallow } from 'enzyme'
import ProviderMock from '../../__mocks__/ProviderMock'
import FeedMock from '../../__mocks__/FeedMock'
import Feed from '../../containers/Feed'

import Header from '../../components/Header'
import Stories from '../../components/Stories'

describe('<Feed />', () => {
	test('Render product', () => {
		const feed = mount(
			<ProviderMock>
				<Feed />
			</ProviderMock>
		)		
		expect(feed.length).toEqual(1)
		expect(feed.html()).toMatchSnapshot()
	})

	test('Comprobar el boton de comprar', () => {
		//const handleAddToCart = jest.fn()
		const feed = mount(
			<ProviderMock>
				<Feed />
			</ProviderMock>
		)
		console.log('Here')
		expect(feed).toHaveLength(1)		
		setImmediate(() => {
			// within `setImmediate` all of the promises have been exhausted    
			feed.update()    			
			expect(feed.find(Header)).toHaveLength(1)
			expect(feed.find(Stories)).toHaveLength(1)
			// have to call `done` here to let Jest know the test is done    
			done()  
		})
		
	})
})