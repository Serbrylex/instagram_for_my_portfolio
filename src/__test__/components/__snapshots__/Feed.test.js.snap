// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feed /> Render product 1`] = `"<div class=\\"sc-bdnxRM hRZjXQ\\"><div class=\\"sc-pNWdM ghWYOG\\"><h1 class=\\"sc-jrsJWt fGxmpT\\">Instagram</h1><div class=\\"sc-ksluID TUyEM\\"><input placeholder=\\"Search\\" class=\\"sc-fnVZcZ ldJCLS\\" value=\\"\\"><a id=\\"search\\" class=\\"sc-hBMUJo jVGdpT\\" href=\\"/search/\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 1024 1024\\" color=\\"white\\" style=\\"color: white;\\" height=\\"22px\\" width=\\"22px\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></a></div><div class=\\"sc-kEqXSa hGnimy\\"><a id=\\"user_picture\\" class=\\"sc-dIsUp liJhhc\\" href=\\"/login/\\"><img src=\\"[object Object]\\" alt=\\"Profile Image\\" class=\\"sc-bqGGPW bVWnqT\\"></a></div></div><div class=\\"sc-cxNHIi fdoaJO\\"><img src=\\"[object Object]\\" alt=\\"loading\\" class=\\"sc-lmgQwP ctTDkf\\"></div><div class=\\"sc-fFSPTT ekUIxE\\"><a class=\\"sc-bkbkJK xqnUt\\" href=\\"/\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 1024 1024\\" color=\\"white\\" style=\\"color: white;\\" height=\\"25px\\" width=\\"25px\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M946.5 505L534.6 93.4a31.93 31.93 0 0 0-45.2 0L77.5 505c-12 12-18.8 28.3-18.8 45.3 0 35.3 28.7 64 64 64h43.4V908c0 17.7 14.3 32 32 32H448V716h112v224h265.9c17.7 0 32-14.3 32-32V614.3h43.4c17 0 33.3-6.7 45.3-18.8 24.9-25 24.9-65.5-.1-90.5z\\"></path></svg></a><a class=\\"sc-bkbkJK xqnUt\\" href=\\"/search\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 1024 1024\\" color=\\"white\\" style=\\"color: white;\\" height=\\"25px\\" width=\\"25px\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z\\"></path></svg></a><a class=\\"sc-bkbkJK xqnUt\\" href=\\"/profile/Serbrylex\\"><img src=\\"[object Object]\\" alt=\\"Profile Image\\" class=\\"sc-iemWCZ gVANCk\\"></a></div></div>"`;
